snap/info.go

```diff
@@ -282,12 +285,17 @@ func (s *Info) CommonDataHomeDir() string {
 	return filepath.Join(dirs.SnapDataHomeGlob, s.Name(), "common")
 }
 
-// UserXdgRuntimeDir returns the XDG_RUNTIME_DIR directory of the snap for a particular user.
+// UserXdgRuntimeDirGlobal returns the toplevel XDG_RUNTIME_DIR directory for a particular user.
+func (s *Info) UserXdgRuntimeDirGlobal(euid int) string {
```

Comment:
If this goes in, it sounds more natural to invert those names as this is the one returning the real XdgRuntimeDir (the one set in the environment variable). The other might become SnapXdgRuntimeDir or similar.